### PR TITLE
Wait for async data in forem mobile cypress test

### PR DESCRIPTION
**Pull request by djuber**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

The forem mobile cypress test is sometimes failing due to userData not being ready (base data request has not returned a response before we check the content, leading to attribute errors on `undefined`).

The [`loginAndVisit(url)`](https://github.com/forem/forem/blob/main/cypress/support/commands.js#L40-L49) command would _normally_ be used for this, but doesn't accept an option argument to pass to the [visitAndWaitForUserSideEffects](https://github.com/forem/forem/blob/main/cypress/support/commands.js#L51-L65) (which does take an option arg, required to stub out the user agent in the request for this test).

The visitAndWaitForUserSideEffects command relies on a utility to intercept the pending user requests, but assumes the user is logged in (and always passes true to this utility method, while for the logged out spec we need to pass it false). 

The second commit (inlining the wait for user side effects logic but passing the [getIntercept function](https://github.com/forem/forem/blob/main/cypress/util/networkUtils.js#L19) a false flag for user logged in to skip waiting for notifications) is the fastest way to do what's necessary, but is a cry for help in deciding what the "real" interface for this looks like - either a visitAndWaitForLoggedOutUserData command is wanted (just like visit and wait for user but passing false), or an extension to the existing method allowing us to flag handling this when we're logged out.

This might be an unusual situation in that we're relying on the async base data (normally for user details) in a context where the user is signed out. I'm not sure if it's worth making it more usable if it's unlikely to be used outside this specific test. 

## QA Instructions, Screenshots, Recordings

Locally, on main, I was seeing failures in both the logged in and logged out contexts when running bin/e2e, at least half of the time. 

On this branch I'm not seeing those issues.

### UI accessibility concerns?

None, test only change.

## Added/updated tests?

- [x] Yes

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: test fix only

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by aitchiss — APPROVED** [review]
>This might be an unusual situation in that we're relying on the async base data (normally for user details) in a context where the user is signed out. I'm not sure if it's worth making it more usable if it's unlikely to be used outside this specific test.

I think the approach you have is fine for this test, and if we find ourselves needing this functionality in future, we can look at refactoring/creating a new helper then 👍

**Comment by djuber** [comment]
This seems like it's still occasionally failing for me in the logged out test (ForemMobile is sometimes undefined, so it's not possible to check getUserData being undefined).

File: cypress/support/commands.js [hunk 1]
```diff
@@ -52,17 +52,23 @@ Cypress.Commands.add('loginAndVisit', (user, url) => {
...
-    cy.visit(url, options);
-  } else {
-    const intercepts = getInterceptsForLingeringUserRequests(url, true);
-    cy.visit(url, options);
-    cy.wait(intercepts);
-  }
-});
+Cypress.Commands.add(
+  'visitAndWaitForUserSideEffects',
+  (url, options, userLoggedIn = true) => {
```
Comment (djuber) [inline cypress/support/commands.js:57]: is this an automatic reformatting based on line length? I didn't do this myself.

**Comment by djuber** [comment]
> I think the approach you have is fine for this test, and if we find ourselves needing this functionality in future, we can look at refactoring/creating a new helper then

I went ahead and added a flag to the existing command - I can revert that commit if it's not needed. I felt like I was breaking a few abstraction boundaries by reaching into the network utils directly to get at the intercept function.

**Review by fdoxyz — APPROVED** [review]
Thanks for these enhancements Dan! 🙌

**Pull request merged**